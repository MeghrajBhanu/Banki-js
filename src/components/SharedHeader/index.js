import classes from "./index.module.css";
import { useSelector } from "react-redux";
const SharedHeader = ({ name, pancard ,flagged}) => {
  const user=useSelector(state=>state.auth.user);
  return (
    
      <section className={classes["shared_header"]}>
        <div className={classes["overlay"]}>
          <div className={classes["container"]}>
            <div className={classes["row"]}>
              <div
                className="col"
                data-wow-delay="0.6s"
                style={{
                  visibility: "visible",
                  animationDelay: "0.6s",
                  animationName: "fadeInUp",
                }}
              >
                <div className={classes["fadeInUp"]}>
                  <h2 className="text-center mb-4">Welcome {name?name:"user"} !</h2>
                  <p className="text-center mt-3">
                    Here are your {flagged} bank accounts linked with {pancard?pancard:user?.pancard}
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    
  );
};
export default SharedHeader;
