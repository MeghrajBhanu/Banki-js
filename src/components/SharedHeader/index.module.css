.shared_header {
    background: url('https://png.pngtree.com/thumb_back/fh260/background/20201009/pngtree-dark-green-cyan-paper-cut-minimalist-background-for-brochure-poster-banner-image_405384.jpg') no-repeat;
    background-size: cover;
    color: #fff;
    display:block;
    
}

.fadeInUp {
    
    margin:auto;
    animation: fadeInUp 1s ease-in;
    animation-fill-mode: both;
}
.overlay {
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.4);
    padding-top: 90px;
    padding-bottom: 90px;
}
@keyframes fadeInUp {
    0% {
        opacity: 0;
        -webkit-transform: translateY(20px);
        -ms-transform: translateY(20px);
        transform: translateY(20px);
    }
    100% {
        opacity: 1;
        -webkit-transform: translateY(0);
        -ms-transform: translateY(0);
        transform: translateY(0);
    }
}

