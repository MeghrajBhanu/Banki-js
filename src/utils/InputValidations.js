export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}