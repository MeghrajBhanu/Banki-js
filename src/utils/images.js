import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;