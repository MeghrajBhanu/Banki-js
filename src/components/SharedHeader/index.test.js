import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import SharedHeader from ".";

describe("SharedHeader", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<SharedHeader />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});