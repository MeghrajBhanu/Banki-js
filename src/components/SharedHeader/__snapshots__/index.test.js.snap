// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedHeader snapshot test 1`] = `
<section
  className="shared_header"
>
  <div
    className="overlay"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
          data-wow-delay="0.6s"
          style={
            Object {
              "animationDelay": "0.6s",
              "animationName": "fadeInUp",
              "visibility": "visible",
            }
          }
        >
          <div
            className="fadeInUp"
          >
            <h2
              className="text-center mb-4"
            >
              Welcome 
              user
               !
            </h2>
            <p
              className="text-center mt-3"
            >
              Here are your 
               bank accounts linked with 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
